File: Examples/VisualBasic/ModifyingAndConvertingImages/ExtractTIFFFramesToBMPImageFormat.vb
Imports Aspose.Imaging.FileFormats.Bmp
Imports Aspose.Imaging.FileFormats.Tiff
Imports Aspose.Imaging.ImageOptions
Imports Aspose.Imaging.Sources

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https://Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages
    Class ExtractTIFFFramesToBMPImageFormat
        Public Shared Sub Run()
            ' ExStart:ExtractTIFFFramesToBMPImageFormat
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_ModifyingAndConvertingImages()

            Using multiImage As TiffImage = DirectCast(Image.Load(dataDir & Convert.ToString("SampleTiff1.tiff")), TiffImage)
                ' Create an instance of int to keep track of frames in TiffImage
                Dim frameCounter As Integer = 0

                ' Iterate over the TiffFrames in TiffImage
                For Each tiffFrame As TiffFrame In multiImage.Frames
                    multiImage.ActiveFrame = tiffFrame

                    ' Load Pixels of TiffFrame into an array of Colors
                    Dim pixels As Color() = multiImage.LoadPixels(tiffFrame.Bounds)

                    ' Create an instance of bmpCreateOptions
                    Dim bmpCreateOptions As New BmpOptions()
                    bmpCreateOptions.BitsPerPixel = 24

                    ' Set the Source of bmpCreateOptions as FileCreateSource by specifying the location where output will be saved
                    bmpCreateOptions.Source = New FileCreateSource(String.Format("{0}\ConcatExtractTIFFFramesToBMP_out{1}.bmp", dataDir, frameCounter), False)

                    ' Create a new bmpImage
                    Using bmpImage As BmpImage = DirectCast(Image.Create(bmpCreateOptions, tiffFrame.Width, tiffFrame.Height), BmpImage)
                        ' Save the bmpImage with pixels from TiffFrame
                        bmpImage.SavePixels(tiffFrame.Bounds, pixels)
                        bmpImage.Save()
                    End Using
                    frameCounter += 1
                Next
            End Using
            ' ExEnd:ExtractTIFFFramesToBMPImageFormat
        End Sub
    End Class
End Namespace

File: Examples/CSharp/ModifyingAndConvertingImages/ExtractTIFFFramesToBMPImageFormat.cs
using Aspose.Imaging.FileFormats.Bmp;
using Aspose.Imaging.FileFormats.Tiff;
using Aspose.Imaging.ImageOptions;
using Aspose.Imaging.Sources;
using System;

/*
This project uses the Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API references
when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information.
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from https://releases.aspose.com/,
install it, and then add its reference to this project. For any issues, questions, or suggestions,
please feel free to contact us using https://forum.aspose.com/
*/

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages
{
    class ExtractTIFFFramesToBMPImageFormat
    {
        public static void Run()
        {
            Console.WriteLine("Running example ExtractTIFFFramesToBMPImageFormat");
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_ModifyingAndConvertingImages();

            using (TiffImage multiImage = (TiffImage)Image.Load(dataDir + "SampleTiff1.tiff"))
            {
                // Create an integer variable to keep track of frames in the TiffImage.
                int frameCounter = 0;

                // Iterate over the TiffFrames in the TiffImage.
                foreach (TiffFrame tiffFrame in multiImage.Frames)
                {
                    multiImage.ActiveFrame = tiffFrame;

                    // Load pixels of the TiffFrame into an array of colors.
                    Color[] pixels = multiImage.LoadPixels(tiffFrame.Bounds);

                    // Create a BmpOptions object.
                    BmpOptions bmpCreateOptions = new BmpOptions();
                    bmpCreateOptions.BitsPerPixel = 24;

                    // Set the source of the BmpOptions to a FileCreateSource that specifies where the output will be saved.
                    bmpCreateOptions.Source = new FileCreateSource(string.Format("{0}\\ConcatExtractTIFFFramesToBMP_out{1}.bmp", dataDir, frameCounter), false);
                    
                    // Create a new BmpImage.
                    using (BmpImage bmpImage = (BmpImage)Image.Create(bmpCreateOptions, tiffFrame.Width, tiffFrame.Height))
                    {
                        // Save the BmpImage with pixels from the TiffFrame.
                        bmpImage.SavePixels(tiffFrame.Bounds, pixels);
                        bmpImage.Save();
                    }
                    frameCounter++;
                }
            }

            Console.WriteLine("Finished example ExtractTIFFFramesToBMPImageFormat");
        }
    }
}